lib/cloner.rb

```diff
@@ -186,21 +186,41 @@ def fetch
     git.remote(remote_name).fetch
   end
 
-  def merge
+  def checkout
     logger.info "Checking out #{branch_name}"
     git.branch(branch_name).checkout
+  end
 
-    logger.info "Merging #{originating_repo}/master into #{branch_name}..."
-    if squash?
-      logger.info 'Squashing!'
-      run_command('git', 'merge', '--squash', "#{remote_name}/master")
-      git.commit(commit_message)
+  def apply_sync_method
+    if sync_method == "squash"
+      squash
+    elsif sync_method == "replace_contents"
+      replace_contents
+    elsif sync_method == "merge"
+      merge
     else
-      logger.info 'Not squashing!'
-      run_command('git', 'merge', "#{remote_name}/master")
+      logger.warn "Invalid sync method #{sync_method}. Merging by default..."
+      merge
     end
   end
 
+  def merge
+    logger.info "Merging #{originating_repo}/master into #{branch_name}..."
+    run_command('git', 'merge', "#{remote_name}/master")
+  end
+
+  def squash
+    logger.info "Squashing #{originating_repo}/master into #{branch_name}..."
+    run_command('git', 'merge', '--squash', "#{remote_name}/master")
+    git.commit(commit_message)
+  end
+
+  def replace_contents
+    logger.info "Committing contents of #{originating_repo}/master into #{branch_name} directly..."
+    commit_id = run_command('git', 'commit-tree', "#{remote_name}/master^{tree}", '-p', 'HEAD', '-m', commit_message)
+    run_command('git', 'update-ref', 'HEAD', commit_id.chomp)
```

Comment:
HEAD is the right thing here because it is itself a pointer to the head of the current branch "branch_name". In this case being more exact (`refs/heads#{branch_name}`) is at best equivalent and probably more likely to be the wrong thing at some point.